aip/general/0193.md

```diff
@@ -34,272 +34,154 @@ is necessary, you **should** provide a link where a reader can get more
-    using either of the following reasons breaches the required
-    formatting and is not allowed: `librariesAreGreat`, `noBooks`.
-
-- `domain string`
-  - The logical grouping to which the `reason` belongs. The error domain
-    is typically the registered service name of the tool or product that
-    generated the error. The domain must be a globally unique value.
-  - *Example*:<br>`pubsub.googleapis.com`
-
-- `metadata map`
-  - Additional structured details about this error, which **should**
-    provide important context for customers to identify resolution
-    steps.  Keys **should** match `/[a-z][a-zA-Z0-9-_]+/`, and be
-    limited to 64 characters in length. When identifying the current
-    value of an exceeded limit, the units **should** be contained in the
-    key, not the value.
-  - *Example*:
-    ```
-    "vmType": "e2-medium",
-    "attachment": "local-ssd=3,nvidia-t4=2",
-    "zone": "us-east1-a"
-    ```
-  - For guidance on using the metadata map, see [Dynamic
-    Variables](#dynamic-variables).
+Bad examples:
 
-##### Dynamic variables
+- `THE_BOOK_YOU_WANT_IS_NOT_AVAILABLE` (overly verbose)
+- `ERROR` (too general)
 
-The best, actionable error messages include dynamic segments. These
-variable parts of the message are specific to a particular request.
-Without such context, it is unlikely that the message will be fully
-actionable by the user.
+The `domain` field is the logical grouping to which the `reason` belongs.
+The error domain  typically the registered service name of the tool or product
+that  generated the error. The domain must be a globally unique value, and is
+typically the service name, e.g. `pubsub.googleapis.com`.
 
-This practice is critical so that machine actors do not need to rely on
-`LocalizedMessage.message`, which is subject to change and is not part
-of the API contract.
+The (reason, domain) pair form a machine-readable way of identifying a particular error.
+Services **must** use the same (reason, domain) pair for the same error, and
+**must not** use the same (reason, domain) pair for logically different errors.
+The decision about whether two errors are "the same" or not is not always clear, but
+should generally be considered in terms of the expected action a client might take
+to resolve them.
 
-Consider the following example:
+The `metadata` field is a map of key/value pairs providing additional
+dynamic information as context. Each key within `metadata` **must** be at most
+64 characters long, and conform to the regular expression `[a-z][a-zA-Z0-9-_]+`.
 
-> The Book, "The Great Gatsby", is unavailable at the Library, "Garfield
-> East".  It is expected to be available again on 2199-05-13.
+Any request-specific information which contributes to the `Status.message` or
+`LocalizedMessage.message` messages **must** be represented within `metadata`.
+This practice is critical so that machine actors do not need to parse error
+messages to extract information.
```

Comment:
Nit: similar to earlier, this reasoning would be best in a Rationale section but I'm not going to be picky

Reply:
Let's leave this open so we can move it and link.

Reply:
Agreed.

Reply:
That's fine with me (punting)